query-algebrizer/src/cc.rs

```diff
@@ -943,6 +979,124 @@ mod testing {
         ]);
     }
 
+    /// This test ensures that we do less work if we know the attribute thanks to a var lookup.
+    #[test]
+    fn test_apply_unattributed_but_bound_pattern_with_returned() {
+        let mut cc = ConjoiningClauses::default();
+        let mut schema = Schema::default();
+        associate_ident(&mut schema, NamespacedKeyword::new("foo", "bar"), 99);
+        add_attribute(&mut schema, 99, Attribute {
+            value_type: ValueType::Boolean,
+            ..Default::default()
+        });
+
+        let x = Variable(PlainSymbol::new("?x"));
+        let a = Variable(PlainSymbol::new("?a"));
```

Comment:
`$` indicates a source variable — a point-in-time DB, the transaction log, etc.

You can supply both as inputs (though in our strongly-typed world, we'll do so through two different collections).